packages/block-editor/src/components/inserter/menu.js

```diff
@@ -122,6 +123,13 @@ function InserterMenu(
 
 	const onInsertPattern = useCallback(
 		( blocks, patternName ) => {
+			if ( blocks.length === 1 && blocks[ 0 ]?.name === 'core/group' ) {
```

Comment:
@MaggieCabrera @ndiego I wonder if this is a technique we could use as a proxy for "section pattern"?

Reply:
Yeah perhaps. You might want to include Cover and maybe Media & Text and Columns.

Reply:
We have a core pattern that is just a cover block, and I think TT4 has one of those too

Reply:
Worth noting that not all blocks support `templateLock`.

Reply:
If that's the case then we'd need to iterate through the blocks applying the correct blockEditingMode. Template Lock works well for a prototype but possible not for production ready code.

Reply:
I've realised that we're actually looking to find the block that is acting as **the root** of all the blocks we conceptually consider to be "sections" on a template/post.

As a result I think it's important that we review https://github.com/WordPress/gutenberg/pull/64544 which might give us a standardised way to compute this.